### PR TITLE
add managed secret creation policy

### PR DESCRIPTION
**Pull request by maidul98**
# Description 📣

We allow users to have the managed kubernetes secret be created in any namespace by default. While this worked for most cases, it didn't for ArgoCD. ArgoCD would not know which resource the managed secret belonged to so it would auto prune it. 

To address the auto prune, we added a resource owner to the managed secret. This resolved the issue temporarily but left users who have managed secrets in namespaces not equal to the InfisicalSecret CRD in a error state https://github.com/Infisical/infisical/issues/1042#issuecomment-1862252370 . To Address both cases, this PR adds creation policy to the managed secret.

Creation polices: `Owner` and `Orphan` (default)

`Owner` will add owner reference to the managed secret 
`Orphan` will set no owner on the managed secret

```
apiVersion: secrets.infisical.com/v1alpha1
kind: InfisicalSecret
metadata:
  name: infisicalsecret-sample
  labels:
    label-to-be-passed-to-managed-secret: sample-value
  annotations:
    reflector.v1.k8s.emberstack.com/reflection-allowed: 'true'
spec:
  hostAPI: https://app.infisical.com/api
  resyncInterval: 10
  authentication:
    serviceToken:
      serviceTokenSecretReference:
        secretName: service-token
        secretNamespace: default
      secretsScope:
        envSlug: dev
        secretsPath: "/"
  managedSecretReference:
    secretName: managed-secret
    secretNamespace: default
    creationPolicy: "Orphan"

``` 

Other small improvements:
 - a small patch was made to prevent auto generated annotations to be propagated  to the managed secret from InfisicalSecret
- hard coded the operator version to prevent users from auto upgrading 
- fixed etags issue where the server doesn't send 304 so we use the etag directly to compare 
- updated helmify in make file so that developer doesn't need to install helmify on their own 

This change is breaking for users who expect owner reference to be set. To set owner reference, users must set managedSecretReference.creationPolicy to "Owner"

## Type ✨

- [x] Bug fix
- [x] New feature
- [x] Breaking change
- [ ] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [ ] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

File: helm-charts/secrets-operator/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
```
Comment (DanielHougaard) [inline helm-charts/secrets-operator/templates/serviceaccount.yaml:1]: Did you mean to include this file?

File: helm-charts/secrets-operator/values.yaml [hunk 2]
```diff
@@ -22,6 +41,8 @@ controllerManager:
         cpu: 10m
         memory: 64Mi
   replicas: 1
+  serviceAccount:
```
Comment (DanielHougaard) [inline helm-charts/secrets-operator/values.yaml:44]: This doesn't seem to have anything to do with the PR, I assume we want to remove this part too?

File: k8-operator/packages/api/api.go [hunk 3]
```diff
@@ -77,7 +77,7 @@ func CallGetSecretsV3(httpClient *resty.Client, request GetEncryptedSecretsV3Req
 		return GetEncryptedSecretsV3Response{}, fmt.Errorf("CallGetSecretsV3: Unsuccessful response. Please make sure your secret path, workspace and environment name are all correct [response=%s]", response)
 	}
 
-	if response.StatusCode() == 304 {
+	if response.Header().Get("etag") == request.ETag {
```
Comment (DanielHougaard) [inline k8-operator/packages/api/api.go:80]: Just FYI this overlaps with my existing K8 PR

**Review by DanielHougaard — CHANGES_REQUESTED** [review]
Awesome! I left a few comments

File: helm-charts/secrets-operator/templates/serviceaccount.yaml (hunk 1 above)
Comment (maidul98) [reply helm-charts/secrets-operator/templates/serviceaccount.yaml:1]: It is apart of the new helmify that was auto generated. It used to exist but now they made it standalone

File: helm-charts/secrets-operator/values.yaml (hunk 2 above)
Comment (maidul98) [reply helm-charts/secrets-operator/values.yaml:44]: Also helmify auto generated

File: k8-operator/packages/api/api.go (hunk 3 above)
Comment (maidul98) [reply k8-operator/packages/api/api.go:80]: Should be a tiny conflict

**Review by DanielHougaard — APPROVED** [review]

**Pull request merged**